src/java/org/apache/cassandra/db/compaction/CompactionTask.java

```
@@ -205,17 +207,24 @@ public boolean apply(SSTableReader sstable)
                             }
                         }
 
-                        // don't replace old sstables yet, as we need to mark the compaction finished in the system table
+                        readyToFinish = true;
                         newSStables = writer.finish();
                     } catch (Exception e) {
-                        throw new CompactionException(taskIdLoggerMsg, ssTableLoggerMsg.toString(), e);
+                        CompactionException exception = new CompactionException(taskIdLoggerMsg, ssTableLoggerMsg.toString(), e);
+                        if (readyToFinish && e.getSuppressed() != null && e.getSuppressed().length != 0)
```

Can a suppressed exception not be null but have a size of 0?

---

And we have confirmed that the only time a suppressed exception is present is when the close statement fails?

---

Throwable should return us an empty array instead of null, but I figure the null check is still nice to have in case this changes between java versions since it's only agreed to in javadoc

---

> And we have confirmed that the only time a suppressed exception is present is when the close statement fails?

This is how suppressed is supposed to be used, yes. In my looking this appears to be the case as well. Ideally even if we accidentally keep the compaction log around it should be relatively infrequent + in theory harmless then removed the next startup